utils/profile-services.ts

```diff
@@ -0,0 +1,185 @@
+import * as a from "fp-ts/lib/Array";
+import { BlockedInboxOrChannelEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/BlockedInboxOrChannel";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { NonNegativeInteger } from "italia-ts-commons/lib/numbers";
+
+const NOT_FOUND_TITLE = "ServicesPreferences not found";
+
+interface IServicePreferenceHandler {
+  readonly isMyReposability: (profile: RetrievedProfile) => boolean;
+  readonly handleProfile: (
+    profile: RetrievedProfile,
+    servicesPreferencesModel: ServicesPreferencesModel,
+    serviceId: ServiceId
+  ) => t.Task<LimitedProfile>;
+}
+
+/**
+ * Converts the RetrievedProfile model to LimitedProfile type.
+ */
+export const retrievedProfileToLimitedProfile = (
+  retrivedProfile: RetrievedProfile,
+  senderAllowed: boolean
+): LimitedProfile => ({
+  preferred_languages: retrivedProfile.preferredLanguages,
+  sender_allowed: senderAllowed
+});
+
+/**
+ * Whether the sender service is allowed to send
+ * messages to the user identified by this profile
+ */
+export const isSenderAllowed = (
+  blockedInboxOrChannels:
+    | RetrievedProfile["blockedInboxOrChannels"]
+    | undefined,
+  serviceId: ServiceId
+): boolean =>
+  blockedInboxOrChannels === undefined ||
+  blockedInboxOrChannels[serviceId] === undefined ||
+  blockedInboxOrChannels[serviceId].indexOf(BlockedInboxOrChannelEnum.INBOX) <
+    0;
+
+const findServicePreference = (
+  servicesPreferencesModel: ServicesPreferencesModel,
+  profile: RetrievedProfile,
+  serviceId: ServiceId
+): te.TaskEither<IResponseErrorNotFound, RetrievedServicePreference> =>
+  te
+    .fromEither(
+      NonNegativeInteger.decode(profile.servicePreferencesSettings.version)
+    )
+    .mapLeft(_ =>
+      ResponseErrorNotFound(
+        NOT_FOUND_TITLE,
+        "preferences settings version must be a natural number"
+      )
+    )
+    .chain(version =>
+      servicesPreferencesModel
+        .find([
+          makeServicesPreferencesDocumentId(
+            profile.fiscalCode,
+            serviceId,
+            version
+          ),
+          profile.fiscalCode
+        ])
+        .mapLeft(_ =>
+          ResponseErrorNotFound(
+            NOT_FOUND_TITLE,
+            "error searching for preference services"
+          )
+        )
+    )
+    .map(
+      e.fromOption(ResponseErrorNotFound(NOT_FOUND_TITLE, "missing preference"))
+    )
+    .chain(te.fromEither);
+
+const servicePreferenceToLimitedProfile = (
+  profile: RetrievedProfile,
+  servicePreference: RetrievedServicePreference
+): LimitedProfile =>
+  retrievedProfileToLimitedProfile(profile, servicePreference.isInboxEnabled);
+
+export const profileWithPreferenceVersionWithModeAuto: IServicePreferenceHandler = {
+  handleProfile: (profile, servicesPreferencesModel, serviceId) =>
+    findServicePreference(servicesPreferencesModel, profile, serviceId).fold(
+      _l => retrievedProfileToLimitedProfile(profile, true),
```

Comment:
Could you add a small comment that explain why we are using `true` in `retrievedProfileToLimitedProfile(profile, true)`?

Reply:
A query error is evaluated as notFound event, and we return `sender_allowed=true` to the service calling `GetLimitedProfile`. Can we add a test on this scenario?